File: language-testing/fibonacci/fibonacci.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <assert.h>

unsigned long fib(int rot) {
  unsigned long old_value = 1;
  unsigned long new_value = 0;
  unsigned long tmp;
  int i;

  for (i = 0; i < rot; ++i) {
    tmp = old_value;
    old_value = new_value;
    new_value = tmp + old_value;
  }
  return new_value;
}

int test() {
  puts("Starting test cases");
  assert(fib(0) == 0);
  assert(fib(1) == 1);
  assert(fib(2) == 1);
  assert(fib(42) == 267914296);
  puts("Success!");
  return 0;
}

int main(int argc, char* argv[]) {
  int i;

  /* Fib the argument, if possible: */
  if (argc == 2) {
    if (!strcmp(argv[1], "test")) {
      return test();
    } else {
      printf("%lu\n", fib(atoi(argv[1])));
    }
  /* Else: run 20 fibs: */
  } else {
    for (i = 0; i < 10; ++i) {
      printf("%lu\t%lu\n", fib(i), fib(i+10));
    }
  }
  return 0;
}

File: language-testing/fibonacci/fibonacci.go
package main

import (
  "os"
  "fmt"
  "strconv"
  "log"
)

func fib (rot int) int {
  old_value := 1
  new_value := 0

  for i := 0; i < rot; i++ {
    old_value, new_value = new_value, new_value + old_value
  }

  return new_value
}

func main() {
  if len(os.Args) == 2 {
    i, err := strconv.Atoi(os.Args[1])
    if err != nil {
      log.Fatal(err);
    }
    fmt.Printf("%d\n", fib(i))

  } else {
    for i := 0; i < 10; i++ {
      fmt.Printf("%d\t%d\n", fib(i), fib(i + 10))
    }
  }
}
